tests/test_response.py

```diff
@@ -11,3 +13,31 @@ def test_response_set_content_type_set():
 def test_response_set_content_type_not_set():
     resp = falcon.Response()
     assert 'content-type' not in resp._headers
+
+
+def test_response_get_headers():
+    resp = falcon.Response()
+    resp.append_header('x-things1', 'thing-1')
+    resp.append_header('x-things2', 'thing-2')
+    resp.append_header('x-things3', 'thing-3')
+
+    headers = resp.headers
+    assert headers['x-things1'] == 'thing-1'
+    assert headers['x-things2'] == 'thing-2'
+    assert headers['x-things3'] == 'thing-3'
+
+
+def test_response_attempt_to_set_read_only_headers():
+    resp = falcon.Response()
+
+    resp.append_header('x-things1', 'thing-1')
```

Comment:
For completeness, could you append multiple values to the same header as well, and check that it comes out as a comma-separated string value?